metricflow/dataflow/builder/dataflow_plan_builder.py

```diff
@@ -125,6 +125,14 @@ class MeasureSpecProperties:
     non_additive_dimension_spec: Optional[NonAdditiveDimensionSpec] = None
 
 
+@dataclass(frozen=True)
+class MetricInputMeasureSpecFilterSet:
```

Comment:
Similar to my other comment - can we rename this? Maybe: `MetricInputMeasureSpecFilterSet` -> `TieredFilterSpecSet`

I also would love to add some more detail to this class for readability. Can we have it take in `measure_filter_specs`, `metric_filter_specs`, and `query_filter_specs`? Then we can just add properties to resolve `all_filter_specs` and `after_aggregation_filter_specs`.
Also maybe call it `after_measure_aggregation_filter_specs` or something similar just to be extra clear, since there are other things that get aggregated in a query besides just measures.

Reply:
Also would be good to add a more descriptive docstring to explain what each tier of filter represents!